atd-vze/src/views/Crashes/Crash.js

```diff
@@ -74,6 +77,20 @@ function Crash(props) {
     return geocoderAddressString;
   };
 
+  const convertGeocoderToName = geocoderID => {
```

Comment:
This certainly works as a method of matching a geocode ID to a geocode name. But I think the better way to do this would be through an association using GraphQL in the Hasura layer. 

The way to do this is to go to the Data tab in Hasura, find the table you want to query and then click the Relationships tab. In our case, we're already querying a single crash by its ID for this view so we want to add a nested relationship from crashes to geocoders via the `geocode_provider` column

![Screen Shot 2019-11-25 at 6 08 23 PM](https://user-images.githubusercontent.com/5697474/69588628-db86eb80-0fae-11ea-9762-c6f602f1deec.png)

Once on that page, scroll to the bottom click "Configure" to add a new relationship manually. 

![Screen Shot 2019-11-25 at 6 10 14 PM](https://user-images.githubusercontent.com/5697474/69588669-f48f9c80-0fae-11ea-9873-db69c5118a4f.png)

The relationship type will be Object since its one to one. Give it a name like "geocode_method". The reference table will be our geocoder lookup table. And then match the related column names. Then click save.

![Screen Shot 2019-11-25 at 6 26 59 PM](https://user-images.githubusercontent.com/5697474/69589385-8ef0df80-0fb1-11ea-814a-4b41b9ceae92.png)

You would now be able to extend the existing `GET_CRASH` query instead of creating a new query and doing a manual join in the client like you currently are...

```graphql
{
  atd_txdot_crashes {
    geocode_method { 
      name
    }
  }
}
```

A couple gotchas... You'll may need to update the `select` permissions for the geocoder table. You also don't want to forget to make the same configuration changes in staging as you will in production. 

This update would also us to start using the geocoding name in the field data lower on the page as well. And in any other feature where we need to pull out the geocoder name instead of the ID in the future.

![Screen Shot 2019-11-25 at 6 21 34 PM](https://user-images.githubusercontent.com/5697474/69589096-759b6380-0fb0-11ea-8286-fca918904fc8.png)

I went ahead and added this relationship to staging. Could you pick it up from here and try to implement this feature to pull data via the crash query and also update the Geocode Provider field? And also make sure the same relationship is configured in production?

Reply:
I like this solution a lot better too. It is much easier to set this up in Hasura and eliminates more JS to maintain. I added this relationship to production and updated the mapping for pulling the geocoder name in PR #477